Index: pom.xml
===================================================================
--- pom.xml	(revision 1130614)
+++ pom.xml	(working copy)
@@ -26,6 +26,7 @@
   </parent>  
   <modelVersion>4.0.0</modelVersion>  
   <artifactId>maven-compile-plugin</artifactId>  
+  <version>1.4.0-incubating-cdm</version>
   <packaging>maven-plugin</packaging>  
   <name>NPanday :: Compile Maven Plugin</name>  
   <description>Maven Plugin for .NET: Handles source and test compiles</description>  
@@ -91,4 +92,17 @@
       </plugin>
     </plugins>
   </build>
+  
+  <distributionManagement>
+    <repository>
+      <id>releases</id>
+      <name>CDM Release Repository </name>
+      <url>https://hudson149.cdm.cdmtech.com/nexus/content/repositories/releases</url>
+    </repository>
+    <snapshotRepository>
+      <id>snapshots</id>
+      <name>CDM Snapshot Repository</name>
+      <url>https://hudson149.cdm.cdmtech.com/nexus/content/repositories/snapshots</url>
+    </snapshotRepository>
+  </distributionManagement>
 </project>
Index: src/main/groovy/npanday/plugin/compile/CompileLifecycleMap.groovy
===================================================================
--- src/main/groovy/npanday/plugin/compile/CompileLifecycleMap.groovy	(revision 1130614)
+++ src/main/groovy/npanday/plugin/compile/CompileLifecycleMap.groovy	(working copy)
@@ -40,25 +40,25 @@
       	def mv_install = "org.apache.maven.plugins:maven-install-plugin:install"
         def mv_deploy = "org.apache.maven.plugins:maven-deploy-plugin:deploy"
 
-        def np_generate_settings = "org.apache.npanday.plugins:NPanday.Plugin.Settings.JavaBinding:$npandayVersion:generate-settings"
-        def np_compile_init = "org.apache.npanday.plugins:maven-compile-plugin:$npandayVersion:initialize"
-        def np_resolve = "org.apache.npanday.plugins:maven-resolver-plugin:$npandayVersion:resolve"
-        def np_generate_assemblyinfo = "org.apache.npanday.plugins:maven-compile-plugin:$npandayVersion:generate-assembly-info"
-        def np_compile_process_sources = "org.apache.npanday.plugins:maven-compile-plugin:$npandayVersion:process-sources"
-        def np_compile_process_test_sources = "org.apache.npanday.plugins:maven-compile-plugin:$npandayVersion:process-test-sources"
-        def np_resgen_copy = "org.apache.npanday.plugins:maven-resgen-plugin:$npandayVersion:copy-resources"
-        def np_resgen_generate = "org.apache.npanday.plugins:maven-resgen-plugin:$npandayVersion:generate"
-        def np_resgen_resx = "org.apache.npanday.plugins:maven-resgen-plugin:$npandayVersion:generate-existing-resx-to-resource"
-        def np_compile = "org.apache.npanday.plugins:maven-compile-plugin:$npandayVersion:compile"
-        def np_test_compile = "org.apache.npanday.plugins:maven-compile-plugin:$npandayVersion:testCompile"
-        def np_test = "org.apache.npanday.plugins:maven-test-plugin:$npandayVersion:test"
-        def np_convert = "org.apache.npanday.plugins:maven-repository-plugin:$npandayVersion:convert-artifact"
+        def np_generate_settings = "org.apache.npanday.plugins:NPanday.Plugin.Settings.JavaBinding:generate-settings"
+        def np_compile_init = "org.apache.npanday.plugins:maven-compile-plugin:initialize"
+        def np_resolve = "org.apache.npanday.plugins:maven-resolver-plugin:resolve"
+        def np_generate_assemblyinfo = "org.apache.npanday.plugins:maven-compile-plugin:generate-assembly-info"
+        def np_compile_process_sources = "org.apache.npanday.plugins:maven-compile-plugin:process-sources"
+        def np_compile_process_test_sources = "org.apache.npanday.plugins:maven-compile-plugin:process-test-sources"
+        def np_resgen_copy = "org.apache.npanday.plugins:maven-resgen-plugin:copy-resources"
+        def np_resgen_generate = "org.apache.npanday.plugins:maven-resgen-plugin:generate"
+        def np_resgen_resx = "org.apache.npanday.plugins:maven-resgen-plugin:generate-existing-resx-to-resource"
+        def np_compile = "org.apache.npanday.plugins:maven-compile-plugin:compile"
+        def np_test_compile = "org.apache.npanday.plugins:maven-compile-plugin:testCompile"
+        def np_test = "org.apache.npanday.plugins:maven-test-plugin:test"
+        def np_convert = "org.apache.npanday.plugins:maven-repository-plugin:convert-artifact"
 
 		def default_validate = [np_compile_init, np_resolve, np_generate_settings]
 		def default_generate_sources = [np_generate_assemblyinfo]
 		def default_process_resources = [np_resgen_copy, np_resgen_generate, np_resgen_resx]
 		def default_process_sources = [np_compile_process_sources, np_compile_process_test_sources]
-		def default_install = [np_convert, "org.apache.npanday.plugins:maven-install-plugin:$npandayVersion:install", mv_install]
+		def default_install = [np_convert, "org.apache.npanday.plugins:maven-install-plugin:install", mv_install]
 		
 		forTypes( [ArtifactType.DOTNET_LIBRARY, ArtifactType.LIBRARY] ) {
 			LifecycleMappingBuilder b->
@@ -81,8 +81,8 @@
 			b.compile (np_compile)
 			b.test_compile (np_test_compile)
 			b.test (np_test)
-			b._package ("org.apache.npanday.plugins:maven-webapp-plugin:$npandayVersion:package")
-			b.deploy ("org.apache.npanday.plugins:maven-webapp-plugin:$npandayVersion:deploy")
+			b._package ("org.apache.npanday.plugins:maven-webapp-plugin:package")
+			b.deploy ("org.apache.npanday.plugins:maven-webapp-plugin:deploy")
 		}
 		forTypes( [ArtifactType.DOTNET_MODULE, ArtifactType.MODULE] ) {
 			LifecycleMappingBuilder b->
@@ -92,7 +92,7 @@
 			b.compile (np_compile)
 			b.test_compile (np_test_compile)
 			b.test (np_test)
-			b._package ("org.apache.npanday.plugins:maven-link-plugin:$npandayVersion:package")
+			b._package ("org.apache.npanday.plugins:maven-link-plugin:package")
 			b.install (default_install)
 			b.deploy (mv_deploy)
 		}
@@ -100,12 +100,12 @@
 			LifecycleMappingBuilder b->
 			b.validate (default_validate)
 			b.generate_sources (default_generate_sources)
-			b.process_sources (np_compile_process_sources, np_compile_process_test_sources, "org.apache.npanday.plugins:maven-aspx-plugin:$npandayVersion:copy-dependency")
+			b.process_sources (np_compile_process_sources, np_compile_process_test_sources, "org.apache.npanday.plugins:maven-aspx-plugin:copy-dependency")
 			b.process_resources (default_process_resources)
-			b.compile (np_compile, "org.apache.npanday.plugins:maven-aspx-plugin:$npandayVersion:compile")
+			b.compile (np_compile, "org.apache.npanday.plugins:maven-aspx-plugin:compile")
 			b.test_compile (np_test_compile)
 			b.test (np_test)
-			b._package ("org.apache.npanday.plugins:maven-aspx-plugin:$npandayVersion:package")
+			b._package ("org.apache.npanday.plugins:maven-aspx-plugin:package")
 			b.install (default_install)
 			b.deploy (mv_deploy)
 		}
@@ -131,7 +131,7 @@
 			b.test_compile (np_test_compile)
 			b.test (np_test)
 			b.install (default_install)
-			b._package ("org.apache.npanday.plugins:maven-mojo-generator-plugin:$npandayVersion:generate-bindings")
+			b._package ("org.apache.npanday.plugins:maven-mojo-generator-plugin:generate-bindings")
             b.deploy (mv_deploy)
 		}
 		forType( ArtifactType.VISUAL_STUDIO_ADDIN ) {
